### PR TITLE
Refactoring consensus feature

### LINKED ISSUE #1374: LAO Object handles features that it should not
The LAO object has too many responsibilities, and they should be decoupled from it.

The remaining features left in the Lao class are :
 - Witnesses
 - Consensus

### PR DESCRIPTION
**Pull request by matteosz**
This PR fixes #1374 by removing the consensus feature from the LAO object, using instead a dedicated repository.

### CODE REVIEW

**Review by pierluca — COMMENTED** [review]
I've reviewed a good chunk of it but I can't say much about some parts. Could you also have a look @simone-kalbermatter ? :)

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ConsensusRepository.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  public void updateElectInstanceByLao(String laoId, @NonNull ElectInstance electInstance) {
+    getLaoConsensus(laoId).updateElectInstance(electInstance);
+  }
+
+  public Optional<ElectInstance> getElectInstance(String laoId, MessageID messageId) {
+    // TODO uncomment that when consensus does not rely on call by reference
+    //    Optional<ElectInstance> optional = lao.getElectInstance(messageId);
+    //    return optional.map(ElectInstance::new); // If empty returns empty optional, if not
+    // returns optional with copy of retrieved ElectInstance
```
Comment (pierluca) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ConsensusRepository.java:59]: outdated comment ?

**Review by pierluca — COMMENTED** [review]
Finally finished the review, minor remarks, otherwise LGTM

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ConsensusRepository.java (hunk 1 above)
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ConsensusRepository.java:59]: Yes it is, I'll update it

**Review by pierluca — APPROVED** [review]

**Pull request merged**